llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -9439,7 +9439,8 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
     }
   }
 
-  if (IsVarArg && Subtarget->isWindowsArm64EC()) {
+  if (IsVarArg && Subtarget->isWindowsArm64EC() &&
+      !(CLI.CB && CLI.CB->isMustTailCall())) {
```

Comment:
I see you have the test case to show that we forward X4 and X5 through musttail. 👍